### ISSUE TITLE
Update github.com/golang/protobuf to v1.3.3

### ISSUE THREAD

**Issue by noahdietz** [bazelbuild/rules_go#2376]
### What version of rules_go are you using?

v0.21.3

### What version of gazelle are you using?

v0.20.0

### What version of Bazel are you using?

2.1.0

### Does this issue reproduce with the latest releases of all the above?

Yes

### What operating system and processor architecture are you using?

Debian x86-64

### What did you do?

Use cloud.google.com/go@v0.53.0 in a project.

go.mod
```
module github.com/noahdietz/repro

go 1.14

require (
	cloud.google.com/go v0.53.0
)
```

WORKSPACE
```
workspace(name = "com_noahdietz_repro")

load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")

http_archive(
    name = "com_google_protobuf",
    strip_prefix = "protobuf-3.11.4",
    urls = ["https://github.com/protocolbuffers/protobuf/archive/v3.11.4.zip"],
    sha256 = "9748c0d90e54ea09e5e75fb7fac16edce15d2028d4356f32211cfa3c0e956564",
)

load("@com_google_protobuf//:protobuf_deps.bzl", "protobuf_deps")

protobuf_deps()

http_archive(
    name = "io_bazel_rules_go",
    urls = [
        "https://storage.googleapis.com/bazel-mirror/github.com/bazelbuild/rules_go/releases/download/v0.21.3/rules_go-v0.21.3.tar.gz",
        "https://github.com/bazelbuild/rules_go/releases/download/v0.21.3/rules_go-v0.21.3.tar.gz",
    ],
    sha256 = "af04c969321e8f428f63ceb73463d6ea817992698974abeff0161e069cd08bd6",
)

load("@io_bazel_rules_go//go:deps.bzl", "go_rules_dependencies", "go_register_toolchains")

go_rules_dependencies()

go_register_toolchains()

http_archive(
    name = "bazel_gazelle",
    urls = [
        "https://storage.googleapis.com/bazel-mirror/github.com/bazelbuild/bazel-gazelle/releases/download/v0.20.0/bazel-gazelle-v0.20.0.tar.gz",
        "https://github.com/bazelbuild/bazel-gazelle/releases/download/v0.20.0/bazel-gazelle-v0.20.0.tar.gz",
    ],
    sha256 = "d8c45ee70ec39a57e7a05e5027c32b1576cc7f16d9dd37135b0eddde45cf1b10",
)

# gazelle:repo bazel_gazelle
load("@bazel_gazelle//:deps.bzl", "gazelle_dependencies", "go_repository")

gazelle_dependencies()

// all of the go_repository rules generated by gazelle follow...
```

BUILD.bazel
```
load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
load("@bazel_gazelle//:def.bzl", "gazelle")

# gazelle:prefix github.com/noahdietz/repro
gazelle(name = "gazelle")

go_library(
    name = "go_default_library",
    srcs = ["main.go"],
    importpath = "github.com/noahdietz/repro",
    visibility = ["//visibility:private"],
    deps = ["@com_google_cloud_go//language/apiv1:go_default_library"],
)

go_binary(
    name = "repro",
    embed = [":go_default_library"],
    visibility = ["//visibility:public"],
)
```

main.go
```go
package main

import (
	"context"
	"fmt"

	language "cloud.google.com/go/language/apiv1"
)

func main() {
	ctx := context.Background()
	c, err := language.NewClient(ctx)
	if err != nil {}
	_ = c
	fmt.Println("Hello, World!")
}
```

### What did you expect to see?

```
$ bazel run :repro
INFO: Analyzed target //:repro (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target //:repro up-to-date:
  bazel-bin/linux_amd64_stripped/repro
INFO: Elapsed time: 0.145s, Critical Path: 0.00s
INFO: 0 processes.
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action
Hello, World!
```

### What did you see instead?

I get a compilation error in `cloud.google.com/go/language/apiv1` because the Go gRPC client stub generated (via golang/protobuf v1.3.2) on the fly expects a concrete type grpc.ClientConn, but is instead receiving an implementation of the grpc.ClientConnInterface from the code in `cloud.google.com/go/language/apiv1`.

```
bazel run :repro
INFO: Analyzed target //:repro (123 packages loaded, 7239 targets configured).
INFO: Found 1 target...
ERROR: ~/.cache/bazel/_bazel_ndietz/4ed524bbb1659f58cebccface4355903/external/com_google_cloud_go/language/apiv1/BUILD.bazel:3:1: GoCompilePkg external/com_google_cloud_go/language/apiv1
/linux_amd64_stripped/go_default_library%/cloud.google.com/go/language/apiv1.a failed (Exit 1) builder failed: error executing command bazel-out/host/bin/external/go_sdk/builder compilepkg -sdk external/go_sdk -ins
tallsuffix linux_amd64 -src external/com_google_cloud_go/language/apiv1/doc.go -src ... (remaining 27 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
compilepkg: error running subcommand: exit status 2
~/.cache/bazel/_bazel_ndietz/4ed524bbb1659f58cebccface4355903/sandbox/linux-sandbox/778/execroot/com_noahdietz_repro/external/com_google_cloud_go/language/apiv1/language_client.go:160:46
: cannot use connPool (type "google.golang.org/api/internal".ConnPool) as type *"google.golang.org/grpc".ClientConn in argument to language.NewLanguageServiceClient
Target //:repro failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 24.691s, Critical Path: 5.14s
INFO: 134 processes: 134 linux-sandbox.
FAILED: Build did NOT complete successfully
FAILED: Build did NOT complete successfully
```

I attempted to use the [Override instructions](https://github.com/bazelbuild/rules_go/blob/master/go/workspace.rst#overriding-dependencies)  but got a bunch of other weird errors. So instead I cloned a local copy of rules_go and wired that up in my WORKSPACE. In the local copy, I changed the version of golang/protobuf being included to v1.3.3. Then everything worked. 

We released the clients in `cloud.google.com/go` with a change to pass a new grpc.ClientConnInterface implementation _after_ `github.com/googleapis/go-genproto` had been regenerated with golang/protobuf v1.3.3. We didn't realize rules_go was at risk here, so apologies on that.

**Comment by linzhp** [comment]
We need this too. Thanks for creating the issue.

**Comment by linzhp** [comment]
@noahdietz You need to follow [this page](https://github.com/bazelbuild/rules_go/wiki/Updating-dependencies) to update the golang/protobuf patch.

**Comment by noahdietz** [comment]
Thanks for the link. Is that what the maintainers want me to do? Or do they have a regular update cycle?

**Comment by linzhp** [comment]
This is what @jayconrod said: https://github.com/bazelbuild/rules_go/pull/2367#issuecomment-585266830

**Comment by jayconrod** [comment]
@noahdietz I'll take a look at this as soon as I can, but between performance reviews and the short week, I have less time than usual.

I suspect that I'll need to cut a new release branch for this, which means everything needs to be updated at the same time. `cloud.google.com/go` definitely needs to keep working.

Preparing a new release takes about a day, assuming there aren't issues with other dependencies.

**Comment by noahdietz** [comment]
Thanks @jayconrod i totally understand. 

Do you want me to open a PR with the dependency update to alleviate some of the work? Or would you rather do it all together?

**Comment by jayconrod** [comment]
Thanks for the offer, but I'd rather do it. Most of the work is updating patches, and reviewing patch changes is hard =P

**Comment by noahdietz** [comment]
Okey dokey, thanks @jayconrod :) LMK if there is anything I can do. The Go cloud client libs team will definitely be more careful about how changes might impact rules_go in the future. Might be worth a quick sync sometime.

**Comment by jayconrod** [comment]
Fixed by #2378

**Comment by jayconrod** [comment]
@noahdietz I'd be happy to do a quick sync any time. The rules_go protobuf integration is not as good as it could be, but I haven't had time to really understand it and make big changes. If you or anyone on your team is able to help improve that, I'd be very grateful. :)
